### PR TITLE
Added wordpress components

### PR DESCRIPTION
**Pull request by guysqr**
*Issue #, if available:*

*Description of changes:*
I added components that can install WordPress in various ways.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Components/Linux/wordpress-recipes-linux/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+phpMyAdmin will be available at `<yourhost>/pma-ui`. 
+
+## User-facing instructions
+
+To use an AMI produced by one of the pipelines in this project you need to create an EC2 instance from the AMI. You will need to configure the instance as you would any other web facing instance. Ensure it has a suitable instance role (eg `AmazonSSMRoleForInstancesQuickSetup` if you want to be able to connect using Systems Manager) and security groups to enable necessary access. 
+
+We advise you NOT to leave access open to the Internet on port 80 - even when testing, you should restrict port 80 to your IP. When in production you should put your instance behind an Application Load Balancer (ALB).
+
+Note that there is no SSL configured on the AMI. We recommend offloading SSL to a ALB and deploying your certificate to that ALB using AWS Certificate Manager (ACM).
```
Comment (austoonz) [inline Components/Linux/wordpress-recipes-linux/README.md:40]: We should not be making service, design or implementation recommendations in this samples repository.

Instead, perhaps move these statements to the top of the readme so they're more obvious, and leave the statements as what isn't being done, and that these samples are not intended for production use.

---

NOTE: These samples are not intended for production use and are provided to demonstrate how WordPress could be installed using EC2 Image Builder. These samples do not implement the best practices guidance for deploying WordPress on AWS (such as ensuring SSL is configured).

File: Components/Linux/wordpress-recipes-linux/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# WordPress ImageBuilder Components
+
+## Variants
+### install_wordpress.yml
+
+1. Assumes php 7.4 is/will be pre-installed (eg from the Amazon-managed `php-7-4-linux` ImageBuilder component)
```
Comment (austoonz) [inline Components/Linux/wordpress-recipes-linux/README.md:6]: I think we should make a statement about the underlying Operating System these are designed for. I am presuming that would be Amazon Linux 2, but we should state this as the samples (using `yum`) clearly won't work on other platforms such as Ubuntu. (I'm not saying we need to make them work elsewhere, just to state what the samples are intended to work on)

File: Components/Linux/wordpress-recipes-linux/install_wordpress.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+name: InstallWordPress
+description: this document installs Wordpress on Amazon Linux
```
Comment (austoonz) [inline Components/Linux/wordpress-recipes-linux/install_wordpress.yml:2]: `This document installs WordPress on Amazon Linux 2` (nits for correct casing and the correct OS)

File: Components/Linux/wordpress-recipes-linux/install_wordpress_with_db.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
+name: InstallWordPressWithDb
+description: this document installs Wordpress with Db on Amazon Linux
```
Comment (austoonz) [inline Components/Linux/wordpress-recipes-linux/install_wordpress_with_db.yml:2]: Similar to above, `This document installs WordPress with a database on Amazon Linux 2`

File: Components/Linux/wordpress-recipes-linux/install_wordpress_with_db_pma.yml [hunk 5]
```diff
@@ -0,0 +1,124 @@
+name: InstallWordPressWithDbPma
+description: this document installs Wordpress with Db on Amazon Linux
```
Comment (austoonz) [inline Components/Linux/wordpress-recipes-linux/install_wordpress_with_db_pma.yml:2]: `This document installs WordPress with a database and phpMyAdmin on Amazon Linux 2`

File: Components/Linux/wordpress-recipes-linux/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+We advise you NOT to leave access open to the Internet on port 80 - even when testing, you should restrict port 80 to your IP. When in production you should put your instance behind an Application Load Balancer (ALB).
+
+Note that there is no SSL configured on the AMI. We recommend offloading SSL to a ALB and deploying your certificate to that ALB using AWS Certificate Manager (ACM).
+
+### Keeping WordPress up-to-date
+
+Note also that while the latest version of WordPress is installed when the AMI is built, you will need to keep WordPress and its plugins and themes up-to-date once you spin up an instance from it. 
+
+We suggest if you spin up new WordPress instances regularly that you schedule the ImageBuilder pipeline to regularly rebuild the AMI so it does not drift too far from the most recent version.
```
Comment (austoonz) [inline Components/Linux/wordpress-recipes-linux/README.md:46]: Perhaps remove `We suggest` and just make this `If you spin up...`.

`We suggest` (IMO) indicates that we're suggesting to continue using these samples for a non-learning type purpose, which is not the intention of these samples (given we have the best practices and reference architecture guides for WordPress on AWS).

**Review by austoonz — CHANGES_REQUESTED** [review]
Thank you for these submissions!

I have added a few comments, primarily around removing statements of design or implementation recommendations, or suggestions that may indicate usage of these samples outside of testing and learning purposes (which is the intention of artifacts in this repository). Instead, for these additions, we should direct customers to our official best practices blog or whitepaper for WordPress on AWS, or to the reference architecture for deploying WordPress on AWS.

Additionally, we may need to include a license summary file, similar to the [one provided](https://github.com/aws-samples/aws-refarch-wordpress#license-summary) in the reference architecture repository. I will need to sync with internal teams about this prior to merging this pull request.

I have not yet performed a proper review of the component contents, I still need to review those. I will most likely clone this and test the components to validate the end state (given we have no automated testing to do that in this repository)

**Comment by guysqr** [comment]
Thanks for those pointers. I will incorporate your suggestions.

**Comment by guysqr** [comment]
It's ready for another review

**Review by austoonz — CHANGES_REQUESTED** [review]
I have reviewed internally and there is no need for additional licensing statements. 

Once the license file and the reference to it included in this PR are removed I'll go ahead and merge.

Thank you!

**Comment by guysqr** [comment]
> I have reviewed internally and there is no need for additional licensing statements.
> 
> Once the license file and the reference to it included in this PR are removed I'll go ahead and merge.
> 
> Thank you!

Hi @austoonz - updates made as per your request. Removed license file and block from readme.

**Pull request merged**